test/test_pager.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+import os
+import requests_mock
+import tableauserverclient as TSC
+
+TEST_ASSET_DIR = os.path.join(os.path.dirname(__file__), 'assets')
+
+GET_XML_PAGE1 = os.path.join(TEST_ASSET_DIR, 'workbook_get_page_1.xml')
+GET_XML_PAGE2 = os.path.join(TEST_ASSET_DIR, 'workbook_get_page_2.xml')
+GET_XML_PAGE3 = os.path.join(TEST_ASSET_DIR, 'workbook_get_page_3.xml')
+
+
+class PagerTests(unittest.TestCase):
+    def setUp(self):
+        self.server = TSC.Server('http://test')
+
+        # Fake sign in
+        self.server._site_id = 'dad65087-b08b-4603-af4e-2887b8aafc67'
+        self.server._auth_token = 'j80k54ll2lfMZ0tv97mlPvvSCRyD0DOM'
+
+        self.baseurl = self.server.workbooks.baseurl
+
+    def test_pager_with_no_options(self):
+        with open(GET_XML_PAGE1, 'rb') as f:
+            page_1 = f.read().decode('utf-8')
+        with open(GET_XML_PAGE2, 'rb') as f:
+            page_2 = f.read().decode('utf-8')
+        with open(GET_XML_PAGE3, 'rb') as f:
+            page_3 = f.read().decode('utf-8')
+        with requests_mock.mock() as m:
+            # Register Pager with default request options
+            m.get(self.baseurl, text=page_1)
+
+            # Register Pager with some pages
+            m.get(self.baseurl + "?pageNumber=1&pageSize=1", text=page_1)
+            m.get(self.baseurl + "?pageNumber=2&pageSize=1", text=page_2)
+            m.get(self.baseurl + "?pageNumber=3&pageSize=1", text=page_3)
+
+            # No options should get all 3
+            workbooks = TSC.Pager(self.server.workbooks)
+            self.assertTrue(len(list(workbooks)) == 3)
+
+            # Let's check that workbook items aren't duplicates
+            workbooks = TSC.Pager(self.server.workbooks)
```

Comment:
generators are one time use.  Once you iterate over them, you cannot go back.  He could have stored the output of `list(workbooks)` and then he wouldn't have had to call it twice.